packages/go_router/lib/src/match.dart

```diff
@@ -3,30 +3,28 @@
 // found in the LICENSE file.
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
 
+import 'matching.dart';
 import 'path_utils.dart';
 import 'route.dart';
 
-/// Each RouteMatch instance represents an instance of a GoRoute for a specific
-/// portion of a location.
+///  An instance of a GoRoute plus information about the current location.
 class RouteMatch {
-  /// Constructor for [RouteMatch], each instance represents an instance of a
-  /// [GoRoute] for a specific portion of a location.
+  /// Constructor for [RouteMatch].
   RouteMatch({
     required this.route,
-    required this.subloc,
-    required this.fullpath,
+    required this.location,
+    required this.template,
```

Comment:
Either we rename every path to template, ie GoRoute.path to GoRoute.template, or we should continue to use path as the name. As far as I know we use path to describe string that contains path parameter, and use location to describe the actual URI string. It was unfortunately we describe both URI.path and URI.toString to both be locations. there should be some distinction between them, but that can be done in a separate pr. For this pr, we should at least try not to create new name.

Reply:
https://github.com/flutter/packages/pull/2453/commits/2712c11e786100746df51938375669334542349f